fix code point mismatch

mhkangg opened the pull request:
**Purpose of this pull request**
This PR fixes the issue #705 a mismatch between defined signature algorithm code points and those actually used in TLS handshakes.

The root cause was that oqsprov_capabilities.c relied on the sigalg_names.fragment template to generate oqs_param_sigalg_list entries only for algorithms with enable_tls=true. As a result, the indices in the generated list no longer aligned with the code points defined in ALGORITHMS.md and generate.yml, leading to incorrect code points being selected during the TLS handshake.

**Summary of changes**

Modified the code generation so that oqs_param_sigalg_list also includes entries for algorithms with enable_tls=false.

To ensure these non-TLS algorithms cannot be matched or used inadvertently in a TLS handshake, their entries are generated with a DISABLED_ prefix.

This preserves the correct index alignment across all algorithms and eliminates the code point mismatch.

**Expected impact**

TLS code points in ClientHello (and other messages) will now correctly match the definitions in ALGORITHMS.md and generate.yml.

Algorithms with enable_tls=false remain unusable for TLS, but their presence in the list prevents index shifting and ensures consistent behavior.

**Additional context**
This resolves the issue where algorithms appended after enable=true & enable_tls=false entries were incorrectly assigned the code point of a preceding algorithm.

baentsch reviewed (approved):
Thanks for the DCO as well as the code fix @mhkangg ! Please feel free to add further input from your side to our project.

mhkangg commented:
Sure, I’ll re-base to main and confirm CI is OK before merge.

mhkangg commented:
Even after rebasing onto main, the CI was still failing, so I checked why, and it turned out that when I modified oqs-template/oqsprov/oqsprov_capabilities.c, entries that had been disabled were also added in oqsprov/oqsprov_capabilities.c. 

However, that file was missing from the commit, which caused the CI test failure. 
I’ve now added the updated oqsprov/oqsprov_capabilities.c, and it looks like the tests are passing.

baentsch commented:
> I just pulled again, merged with main, and it looks like the conflicts are now resolved.

They are - Thanks for that -- but now the formatting broke. Did you [run ./scripts/format_code.sh](https://github.com/open-quantum-safe/oqs-provider/blob/main/CONTRIBUTING.md#coding-style) before commit?

mhkangg commented:
Oh, let me run ./scripts/format_code.sh to check and then commit again.

baentsch commented:
Thanks for the code formatting update @mhkangg -- that test now passed, but when running full CI, many OV tests now fail (see logs)... :-(

mhkangg commented:
It seems there are issues with QEMU cross-compilation and entries with enable_tls=false not being properly disabled. I’ll check and fix them.

baentsch commented:
I'm afraid it's not just in cross-compilation: All `linux_intel` tests also fail in the same way

```
5: Cert/keygen failed for DISABLED_OV_Is_pkc at /__w/oqs-provider/oqs-provider/_build/test/tmp/DISABLED_OV_Is_pkc_srv.crt//__w/oqs-provider/oqs-provider/_build/test/tmp/DISABLED_OV_Is_pkc_srv.key
5:   TLS-SIG handshake test failed: DISABLED_OV_Is_pkc, return code: -1
5: 0090C9C45F7F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:355:Non-default library context, Algorithm (DISABLED_OV_Is_pkc : 0), Properties (provider=oqsprovider)
```
What irks me is the algorithm name displayed: It doesn't really make sense: It seems like `OSSL_PROVIDER_query_operation(oqsprov, OSSL_OP_SIGNATURE` (driving the test) delivers some strange algorithm names in response....

baentsch commented:
@ashman-p : Can you please explain how this flag "enable_tls" that causes this problem actually made it into the code base and how it got tested to not cause problems (of the sort this PR tries to fix)? I do not find a PR for it, just [a commit](https://github.com/open-quantum-safe/oqs-provider/commit/03e4540aba3e7fb326a8ffa95c5b6e9cc3131377) ... This happened right after you assumed [the maintainership role](https://github.com/open-quantum-safe/oqs-provider/pull/664) for the project so one has to assume you validated this approach before merging it, right? Can you please point to the PR containing this commit as well as the discussions around it so I can understand the logic behind this? Or did this wind up in the code base without any review? If so, how? As we're fortunately in the same location right now, would you have time to discuss also this in person, Norm? 

Also tagging @dstebila who may have had an in-person discussions with @praveksharma on this before he left: Any input would be welcome to understand this such as to provide a sensible fix or alternative approach.

dstebila commented:
> Also tagging @dstebila who may have had an in-person discussions with @praveksharma on this before he left: Any input would be welcome to understand this such as to provide a sensible fix or alternative approach.

Sorry, I don't have any insight on this.

bhess commented:
> I do not find a PR for it, just [a commit](https://github.com/open-quantum-safe/oqs-provider/commit/03e4540aba3e7fb326a8ffa95c5b6e9cc3131377)

I think this was part of https://github.com/open-quantum-safe/oqs-provider/pull/670, which disabled some algorithms for TLS because they exceed OpenSSL limits.

RodriM11 commented:
PD: I have created #711 to include the proposed fix to the bug.
